.github/workflows/build.yaml

```diff
@@ -201,11 +206,158 @@ jobs:
+          rm ./*.deb
+      - name: checkout reference-contracts
+        uses: actions/checkout@v3
+        with:
+          repository: AntelopeIO/reference-contracts
+          path: reference-contracts
+      - name: Build & Test reference-contracts
+        run: |
+          cmake -S reference-contracts -B reference-contracts/build -DCMAKE_BUILD_TYPE=Release -DBUILD_TESTS=On
+          cmake --build reference-contracts/build -- -j $(nproc)
+          cd reference-contracts/build/tests
+          ctest --output-on-failure -j $(nproc)
+
+  libtester-build-tree-test:
+    name: libtester build tree test
+    needs: [d, Build]
+    if: always() && needs.Build.result == 'success'
+    strategy:
+      fail-fast: false
+      matrix:
+        platform: [ubuntu20, ubuntu22]
+    runs-on:  ["self-hosted", "enf-x86-midtier"]
+    container: ${{fromJSON(needs.d.outputs.p)[matrix.platform].image}}
+    steps:
+      - name: Clone leap
+        uses: actions/checkout@v3
+        with:
+          submodules: recursive
+      - name: Download leap builddir
+        uses: actions/download-artifact@v3
+        with:
+          name: ${{matrix.platform}}-build
+      - name: checkout reference-contracts
+        uses: actions/checkout@v3
+        with:
+          repository: AntelopeIO/reference-contracts
+          path: reference-contracts
+      - name: Extract leap build
+        run: |
+          zstdcat build.tar.zst | tar x
+          echo "leap_DIR=$PWD/build/lib/cmake/leap" >> "$GITHUB_ENV"
+      - name: Download cdt
+        uses: AntelopeIO/asset-artifact-download-action@v2
+        with:
+          owner: AntelopeIO
+          repo: cdt
+          file: 'cdt_.*amd64.deb'
+          target: main
+          artifact-name: cdt_ubuntu_package_amd64
+          token: ${{github.token}}
+      - name: Install cdt Packages
+        run: |
+          apt install -y ./*.deb
+          rm ./*.deb
+      - name: Build & Test reference-contracts
+        run: |
+          cmake -S reference-contracts -B reference-contracts/build -DCMAKE_BUILD_TYPE=Release -DBUILD_TESTS=On
+          cmake --build reference-contracts/build -- -j $(nproc)
+          cd reference-contracts/build/tests
+          ctest --output-on-failure -j $(nproc)
+
   all-passing:
     name: All Required Tests Passed
-    needs: [dev-package, tests, np-tests]
+    needs: [dev-package, tests, np-tests, libtester-build-tree-test, libtester-make-dev-install-test, libtester-deb-install-test]
     if: always()
     runs-on: ubuntu-latest
     steps:
-      - if: needs.dev-package.result != 'success' || needs.tests.result != 'success' || needs.np-tests.result != 'success'
+      - if: needs.dev-package.result != 'success' || needs.tests.result != 'success' || needs.np-tests.result != 'success' ||
+            needs.libtester-build-tree-test.result != 'success' || needs.libtester-make-dev-install-test.result != 'success' || needs.libtester-deb-install-test.result != 'success'
```

Comment:
I know I postulated this change in a previous commit, but from what I can tell:
* `midtier` didn't really make the build all that much faster than `lowtier`; the cost difference is very minor between low vs mid, so it's not something I'm going to fret about, but
* it seems like requiring libtester jobs is further pushing the required CI run time out. We're moving in the wrong direction here,
  * 3.2: Tests+NP tests, ~4m
  * main: Tests+NP tests, ~5.5m
  * this PR: Tests+NP tests+libtester, ~7m it seems?
  * and then #1142 would likely push a full Build+Tests+NP tests+libtester out another ~4-6m :disappointed:

I'm a bit in conflict what to do here... it's an important test but we need to starting holding the line on these times somehow.

Reply:
So, one of the issues is that there is a new critical path through the tests which adds one step to the dependency tree.  The deb install test is dependent on the leap-dev package.  The long pole right now, is `NP Tests` which take ~5m21s to complete.  In that same time you now have `leap-dev` ~1m55s and then `libtester deb install` at ~4m37s.  That gives you a net increase of ~1m11s in run time and then spin-up time for the jobs.  We could look at only making the success dependent on the libtester scenarios that can be run in parallel.

Reply:
![image](https://github.com/AntelopeIO/leap/assets/11048242/dc602f89-b659-4eee-9151-e633165b89f5)